compiler/src/main/java/bullet/impl/ComponentProcessingStep.java

```diff
@@ -145,6 +150,15 @@ private String getName(TypeMirror type) {
 
     final ClassName elementName = ClassName.get(element);
 
+    // Filter out duplicate injections
```

Comment:
If you have several methods injecting the same type, then you're doing it wrong, and it shouldn't matter if the code is suboptimal (or in this case which exact method, and which kind of method is used).